frontend/catalyst/qjit_device.py

```diff
@@ -118,7 +124,23 @@ def _set_supported_operations(config):
 
         if QJITDevice._check_quantum_control(config):  # pragma: nocover
             # TODO: Once control is added on the frontend.
-            ...
+            gates_to_be_decomposed_if_controlled = [
+                "Identity",
+                "CNOT",
+                "CY",
+                "CZ",
+                "CSWAP",
+                "CRX",
+                "CRY",
+                "CRZ",
+                "CRot",
+            ]
+            native_controlled_gates = [
+                f"C({gate})"
+                for gate in native_gates
+                if gate not in gates_to_be_decomposed_if_controlled
+            ]
+            QJITDevice.operations += native_controlled_gates
```

Comment:
I think this works for now, but I was hoping we would overwrite the `supports_operation(self, operation)` function of the device to provide a more flexible method of telling PL whether a particular gate is supported or not, something like the following:
```py
        def supports_operation(self, operation):
            """Overwrite base class implementation to enact more control over supported gates."""

            if operation[0:2] == "C(":
                operation = operation[2:-1]

            return super().supports_operation(operation)
```
Admittedly the fact that it acts on a string representation of the operator is not great, so maybe it would be better to wait until the new device API, but this is what I had in mind to enforce the TOML spec. Let me know what you think :)

Reply:
In the default 'supports_operation' method [we can see](https://github.com/PennyLaneAI/pennylane/blob/95129a0d6365b48cb4acfa828ceb6a8532e47ef5/pennylane/_device.py#L925) the `Device.operations` check, so modifying this filed is a more generic approach, at least for now.